openedx/features/calendar_sync/models.py

```diff
@@ -0,0 +1,40 @@
+"""
+Models for Calendar Sync
+"""
+from django.contrib.auth.models import User
+from django.db import models
+from simple_history.models import HistoricalRecords
+
+from opaque_keys.edx.django.models import CourseKeyField
+
+
+class UserCalendarSyncConfig(models.Model):
+    """
+    Model to track if a user has the calendar integration enabled for a specific Course
+
+    .. no_pii:
+    """
+    user = models.ForeignKey(User, db_index=True, on_delete=models.CASCADE)
+    course_key = CourseKeyField(max_length=255, db_index=True)
+    enabled = models.BooleanField(default=False)
+
+    history = HistoricalRecords()
+
+    class Meta:
+        unique_together = ('user', 'course_key',)
+
+    @classmethod
+    def is_enabled_for_course(cls, user, course_key):
+        """
+        Check if the User calendar sync is enabled for a particular course.
+        Returns False if the object does not exist.
+
+        Parameters:
+            user (User): The user to check against
+            course_key (CourseKey): The course key to check against
+        Returns:
+            (bool) True if the config exists and is enabled. Otherwise, False
+        """
+        if not cls.objects.exists(user=user, course_key=course_key):
+            return False
+        return cls.objects.get(user=user, course_key=course_key).enabled
```

Comment:
This is 2 lookups, yeah? Try/except might get you to just one.

Reply:
You got it dude